ANN.py

```diff
@@ -24,26 +25,77 @@ def defineOutput(outputNames):
-with open('IRIS.csv', newline='') as csvfile:
-    datasetReader = csv.reader(csvfile, delimiter=',', quotechar='|')
-    data = [row for row in datasetReader]
-#strip data
-output = [data[index][4] for index, name in enumerate(data)]
-output = output[1:] # get ride of top labels
-output = [name for name in set(output)]#remove duplicates
-outputId = defineOutput(output)#set every output to a id
-data = data[1:]#remove label row
-# replace output with a output Id
-for index, name in enumerate(data):
-    for jndex, id in enumerate(outputId):
-        data[index][4] = outputId[jndex] if data[index][4] == output[jndex] else  data[index][4]
-#def createIOarrays:
     #highest number (greater than 0) is use to divide the dataset
     #if there are negative numbers add everything but the largest negative numbers
-    #change all output to number array of 1, 0
 
+def NormilizeData(data):
+    maxNum = 0.0
+    minNum = 0.0
+
+    #find largest negative number
+    for listItem in data:
+        for item in listItem:
+            try:
+                compar = float(item)
+                if compar < minNum:
+                    minNum = compar
+            except:
+                #if it checks th id, pass
+                pass
+
+    # add every number by that negative
+    if minNum < 0:
+        for index, listItem in enumerate(data):
+            for jndex, item in enumerate(listItem):
+                try:
+                    compar = float(item)
+                    data[index][jndex] = compar - minNum
+                except:
+                    pass
+
+#find largest positive number
+    for listItem in data:
+        for item in listItem:
+            try:
+                compar = float(item)
+                if compar > maxNum:
+                    maxNum = compar
+            except:
+                #if it checks th id, pass
+                pass
+
+    #normilize dataset
+    for index, listItem in enumerate(data):
+        for jndex, item in enumerate(listItem):
+            try:
+                compar = float(item)
+                data[index][jndex] = compar / maxNum
+            except:
+                pass
+
+    for i in data:
+        print(i)
+    #read the csv file
+def csvReader(fname):
+    with open(fname, newline='') as csvfile:
+        datasetReader = csv.reader(csvfile, delimiter=',', quotechar='|')
+        data = [row for row in datasetReader]
+    #strip data
+    output = [data[index][4] for index, name in enumerate(data)]
+    output = output[1:] # get ride of top labels
+    output = [name for name in set(output)]#remove duplicates
+    outputId = defineOutput(output)#set every output to a id
+    data = data[1:]#remove label row
+    # replace output with a output Id
+    for index, name in enumerate(data):
+        for jndex, id in enumerate(outputId):
+            data[index][4] = outputId[jndex] if data[index][4] == output[jndex] else  data[index][4]
+
+    NormilizeData(data)
+
+csvReader("IRIS.csv")
+
+"""

```

Comment:
Maybe remove this bulk comment before commiting the changes, But otherwise the general flow of the above sections seems good (I havent run any actual testing on it as im assuming that you have, and we will be doing more testing in the coming weeks)